File: 10828.cpp
#include <iostream>
#include <stack>

using namespace std;

int main(){
    int n,r;
    stack<int> arr;
    cin >> n;

    for(int i=0; i<n; i++){
        string cmd;
        int num;

        cin >> cmd;

        if(cmd == "push"){
            cin >> num;
            arr.push(num);
        }
        else if(cmd == "top"){
            if(arr.size() > 0){
                r = arr.top();
            }
            else{
                r = -1;
            }
            cout << r <<"\n";

        }
        else if(cmd == "size"){
            r = arr.size();
            cout << r <<"\n";

        }
        else if(cmd == "empty"){
            if(arr.size() == 0){
                r = 1;
            }
            else{
                r = 0;
            }
            cout << r <<"\n";
        }
        else if(cmd == "pop"){
            if(arr.size()==0){
                r = -1;
                cout << r <<"\n";
            }
            else{
                r = arr.top();
                arr.pop();
                cout << r <<"\n";
            }
        }
    }
}

File: 10828.py
import sys

command = ['push','pop','size','empty','top']

n = int(sys.stdin.readline())
stack = []

for _ in range(n):
    com = sys.stdin.readline().split()

    if command[0] == com[0]:
        stack.append(int(com[1]))
    elif command[1] == com[0]:
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif command[2] == com[0]:
        print(len(stack))
    elif command[3] == com[0]:
        if stack:
            print(0)
        else:
            print(1)
    elif command[4] == com[0]:
        if stack:
            print(stack[-1])
        else:
            print(-1)
